### PR TITLE
chore(macros/HTTPSidebar): Reorg HTTP sidebar after doc reshuffle, delete 'Basics of HTTP'

### PR DESCRIPTION
**Pull request by bsmth**
## Summary

This PR reorganizes the HTTP sidebar after a reshuffle and reorg of docs under HTTP.

__dependencies:__

Should be merged after:

- [x] https://github.com/mdn/content/pull/35774

### Problem

'Basics of HTTP' doc is proposed deletion. The index also surfaces some guides that are hidden further down the tree.

### Solution

Restructure the HTTP sidebar to better reflect IA of HTTP section.

### Before

image

### After

image

## How did you test this change?

* `yarn && yarn dev`
* http://localhost:3000/en-US/docs/Web/HTTP

### CODE REVIEW

File: kumascript/macros/HTTPSidebar.ejs [hunk 1]
```diff
@@ -292,56 +313,55 @@ var text = mdn.localStringMap({
...
-            <summary><%=text['HTTPGuide']%></summary>
-            <ol>
-                <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Basics_of_HTTP`, null, text['Basics'])%></li>
                 <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Overview`, null, text['Overview'])%></li>
-                <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Basics_of_HTTP/Evolution_of_HTTP`, null, text['Evolution'])%></li>
                 <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Messages`, null, text['Messages'])%></li>
                 <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Session`, null, text['Session'])%></li>
+                <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Compression`, null, text['Compression'])%></li>
+                <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Content_negotiation`, null, text['ContentNego'])%></li>
+                <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Content_negotiation`, null, text['Proxies'])%></li>
```
Comment (Josh-Cena) [inline kumascript/macros/HTTPSidebar.ejs:323]: Wrong one?

**Comment by Josh-Cena** [comment]
1. I think there should be a single link to the whole "URI" doc instead of many. WDYT?
2. https://jc-verse.github.io/mdn-graph/warnings reports a few other unreachable pages. Not sure if you want to add them too.
  - [/en-us/web/http/browser_detection_using_the_user_agent](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent)
  - [/en-us/web/http/configuring_servers_for_ogg_media](https://developer.mozilla.org/en-US/docs/Web/HTTP/Configuring_servers_for_Ogg_media)
  - [/en-us/web/http/cross-origin_resource_policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cross-Origin_Resource_Policy)
  - The landing pages of `Headers/`, `Methods/`, and `Status/`
  - [/en-us/web/http/network_error_logging](https://developer.mozilla.org/en-US/docs/Web/HTTP/Network_Error_Logging)
  - [/en-us/web/http/proxy_servers_and_tunneling](https://developer.mozilla.org/en-US/docs/Web/HTTP/Proxy_servers_and_tunneling)

File: kumascript/macros/HTTPSidebar.ejs (hunk 1 above)
Comment (bsmth) [reply kumascript/macros/HTTPSidebar.ejs:323]: Whoops, copy paste err

**Comment by bsmth** [comment]
> 1. I think there should be a single link to the whole "URI" doc instead of many. WDYT?

I'll take a look how this appears, but I think you are probably right.

>https://jc-verse.github.io/mdn-graph/warnings reports a few other unreachable pages. Not sure if you want to add them too. 

Very convenient, ty. I think these should be in there as long as they live in this directory (Ogg guide may be a candidate to be moved out to https://developer.mozilla.org/en-US/docs/Web/Media)

File: kumascript/macros/HTTPSidebar.ejs [hunk 2]
```diff
@@ -291,93 +312,84 @@ var text = mdn.localStringMap({
...
+          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Headers`, null, text['Headers'])%></li>
+          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Methods`, null, text['Methods'])%></li>
+          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Status`, null, text['Status'])%></li>
+          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Session`, null, text['Session'])%></li>
+          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Compression`, null, text['Compression'])%></li>
+          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Content_negotiation`, null, text['ContentNego'])%></li>
+          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Proxy_servers_and_tunneling`, null, text['Proxies'])%></li>
+          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Connection_management_in_HTTP_1.x`, null, text['Connection1x'])%></li>
+          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Protocol_upgrade_mechanism`, null, text['ProtocolUpgradeMech'])%></li>
+          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Evolution_of_HTTP`, null, text['Evolution'])%></li>
```
Comment (caugner) [inline kumascript/macros/HTTPSidebar.ejs:329]: Can we remove all the smartLink parameters except the first one? It'll automatically draw `short-title || title` from the page.

Comment (bsmth) [reply kumascript/macros/HTTPSidebar.ejs:329]: Do you mean all smartLinks should have the following format?

```
<ol>
  <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Overview`)%></li>
  <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Messages`)%></li>
  <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Headers`)%></li>
  <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Methods`)%></li>
  <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Status`)%></li>
  <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Session`)%></li>
  <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Compression`)%></li>
  <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Content_negotiation`)%></li>
  <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Proxy_servers_and_tunneling`)%></li>
  <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Connection_management_in_HTTP_1.x`)%></li>
  <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Protocol_upgrade_mechanism`)%></li>
  <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Evolution_of_HTTP`)%></li>
</ol>
```

Comment (caugner) [reply kumascript/macros/HTTPSidebar.ejs:329]: Yes, exactly. This is how it will be in the new sidebar architecture.

Comment (bsmth) [reply kumascript/macros/HTTPSidebar.ejs]: @caugner do you mind having another look?

File: kumascript/macros/HTTPSidebar.ejs [hunk 3]
```diff
@@ -280,104 +290,88 @@ var text = mdn.localStringMap({
...
+      <details>
+        <summary><%=text['Security']%></summary>
+        <ol>
+          <li><%-web.smartLink(`/${locale}/docs/Web/Security/Practical_implementation_guides`)%></li>
+          <li><a href="/en-US/observatory">HTTP Observatory</a></li>
+          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Permissions_Policy`)%></li>
+          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/CSP`)%></li>
+          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/CORS`)%></li>
+          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Cross-Origin_Resource_Policy`)%></li>
+          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Headers/Strict-Transport-Security`, null, "HTTP Strict Transport Security (HSTS)")%></li>
```
Comment (caugner) [inline kumascript/macros/HTTPSidebar.ejs:334]: Curious why we're setting a label explicitly here?

```suggestion
          <li><%-web.smartLink(`/${locale}/docs/Web/HTTP/Headers/Strict-Transport-Security`)%></li>
```

File: kumascript/macros/HTTPSidebar.ejs [hunk 4]
```diff
@@ -15,16 +15,14 @@ function state(section) {
 var text = mdn.localStringMap({
   'en-US': {
     'HTTP': 'HTTP',
-    'HTTPGuide': 'HTTP guide',
-    'Basics': 'Basics of HTTP',
     'Overview': 'Overview of HTTP',
     'Evolution': 'Evolution of HTTP',
-    'ResourcesURI': 'Resources and URIs',
+    'URIsResources': 'URIs and resources',
     'Identifying': 'Identifying resources on the Web',
```
Comment (caugner) [inline kumascript/macros/HTTPSidebar.ejs:21]: I think there's a lot of keys here no longer needed due to using the page's `short-title || title`. Essentially we only need the section titles, right?

Comment (bsmth) [reply kumascript/macros/HTTPSidebar.ejs:21]: There are some exceptions in the `<details>` elements:

https://github.com/mdn/yari/pull/11753/files#diff-2b04bb91aa3c362eca410b8ef5a10471dcb5fa9e6672e3ee6c9d7c934d300468L359

I can remove all the keys except those? I also don't want to leave it half-refactored, but I'm happy to follow your lead here

Comment (caugner) [reply kumascript/macros/HTTPSidebar.ejs:21]: Yup, you can remove all that aren't referenced.

File: kumascript/macros/HTTPSidebar.ejs (hunk 2 above)
Comment (caugner) [reply kumascript/macros/HTTPSidebar.ejs]: Looking good!

File: kumascript/macros/HTTPSidebar.ejs (hunk 4 above)
Comment (bsmth) [reply kumascript/macros/HTTPSidebar.ejs]: done in https://github.com/mdn/yari/pull/11753/commits/1d9ff293ce5748e194049b319b59e7ab56b4f624

**Review by caugner — APPROVED** [review]

**Comment by bsmth** [comment]
Thank you! Merging shortly

**Pull request merged**